docs/enhanced-depends-logic.md

```diff
@@ -25,7 +25,10 @@ available task results is as follows:
 | `.Omitted` | Task Omitted | Task was omitted |
 | `.Daemoned` | Task is Daemoned and is not Pending | |
 
-For convenience, if an omitted task result is equivalent to `(task.Succeeded || task.Skipped || task.Daemoned)`.
+A tasks is considered `Skipped` if its `when` condition evaluates to false. On the other hand, if a task doesn't run
+because its `depends` evaluated to false it is `Omitted`.
```

Comment:
```suggestion
A task is `Skipped` if its `when` condition evaluated to false. A task is `Omitted` if its `depends` condition evaluated to false.
```

- grammar: singular "A task", not "A tasks"
- grammar: use consistent tense -- "evaluates" + "evaluated" -> "evaluated"
  - past tense is also correct in this case, the task result is after the task.
- simplify, [per k8s style guide](https://kubernetes.io/docs/contribute/style/style-guide/#use-simple-and-direct-language)
  - also "On the other hand" is an idiom that should be avoided, per [k8s style guide](https://kubernetes.io/docs/contribute/style/style-guide/#avoid-jargon-and-idioms)
    - and it is not quite accurate either; it typically refers to opposites or contrasts, whereas `depends` and `when` can be used together and are independent.
    
Also, rather than having this as two separate sentences below the table, these should just be directly in the table.

Reply:
all of the above completed in #13110

Reply:
Nice. Yes, #13110 is a better fix than mine.